lib/semian/mysql2.rb

```diff
@@ -53,11 +55,19 @@ def semian_identifier
     end
 
     def query(*args)
-      acquire_semian_resource(adapter: :mysql, scope: :query) { raw_query(*args) }
+      if query_whitelisted?(*args)
+        raw_query(*args)
+      else
+        acquire_semian_resource(adapter: :mysql, scope: :query) { raw_query(*args) }
+      end
     end
 
     private
 
+    def query_whitelisted?(sql, *)
+      QUERY_WHITELIST.any? { |pattern| sql =~ pattern }
```

Comment:
I think the regexp should be fine. My **ass**umption is that the anchor at the beginning of both `QUERY_WHITELIST` expressions would effectively force them to run in constant-time, regardless of the size of the input string.